plugins/inputs/linux_cpu/linux_cpu_test.go

```diff
@@ -0,0 +1,98 @@
+//go:build linux
+
+package linux_cpu
+
+import (
+	"github.com/influxdata/telegraf/testutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNoMetrics(t *testing.T) {
+	plugin := &LinuxCPU{}
+	require.Error(t, plugin.Init())
+}
+
+func TestNoCPUs(t *testing.T) {
+	td := t.TempDir()
+
+	plugin := &LinuxCPU{
+		Metrics:   []string{"cpufreq"},
+		PathSysfs: td,
+	}
+	require.Error(t, plugin.Init())
+}
+
```

Comment:
Would be also nice to test scenario when `discoverCpus()` returns no error and empty lists (it would be for case when directories for cpus exist but all properties cannot be loaded).